lib/modules/manager/puppet/puppetfile-parser.ts

```diff
@@ -0,0 +1,86 @@
+import { newlineRegex, regEx } from '../../../util/regex';
+import type { PuppetForgeUrl, Puppetfile, PuppetfileModule } from './types';
+
+const forgeRegex = regEx(/^forge\s+['"]([^'"]+)['"]/);
+
+export function parsePuppetfile(content: string): Puppetfile {
```

Comment:
This is probably a good candidate to create a `Puppetfile` class amd add required functionallity to it 🙃 
I think that would make usage simpler ans clearer.